-1

**Failed Tests**:   Build  ClangBuild 
**Size**: This PR adds an extra 24KB to repository
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ca51ff/41972/summary.html
**COMMIT**: d279605a8a3648b798f6deda762ea90cdf4dff9d
**CMSSW**: CMSSW_14_2_X_2024-10-04-1100/el8_amd64_gcc12
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week1/cms-sw/cmssw/46262/41972/install.sh` to create a dev area with all the needed externals and cmssw changes.



The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
 - @smuzaffar cms-sw/cmssw#46208
 - @smuzaffar cms-sw/cmssw#46219
 - @duff-ae cms-sw/cmssw#46231
 - @smuzaffar cms-sw/cmssw#46212
 - @smuzaffar cms-sw/cmssw#46213
 - @smuzaffar cms-sw/cmssw#46211
 - @smuzaffar cms-sw/cmssw#46205
 - @smuzaffar cms-sw/cmssw#46214
 - @iarspider cms-sw/cmssw#46226
 - @smuzaffar cms-sw/cmssw#46232
 - @battibass cms-sw/cmssw#46224
 - @smuzaffar cms-sw/cmssw#46239
 - @denizsun cms-sw/cmssw#46173
 - @smuzaffar cms-sw/cmssw#46238
 - @smuzaffar cms-sw/cmssw#46233

You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ca51ff/41972/git-recent-commits.json
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-ca51ff/41972/git-merge-result


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin src/IORawData/SiPixelInputSources/src/PixelSLinkDataInputSource.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_14_2_X_2024-10-04-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_14_2_X_2024-10-04-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-04-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/pcre/8.43-e34796d17981e9b6d174328c69446455/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/boost/1.80.0-477823d53efabc5118f199265eb7ab49/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/lcg/root/6.30.09-c59d6b036f1cbd6988c172ba319259f1/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/tbb/v2021.9.0-e785b749a0b6cb9c66dc1d78066210e0/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/zlib/1.2.11-1a082fc322b0051b504cc023f21df178/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/fmt/10.2.1-deba157dd76fed17ac8b42adc60ff53f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/tinyxml2/6.2.0-68d511689c5dbd9a0d296bbb795fa052/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v2 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/IORawData/SiPixelInputSources/src/IORawDataSiPixelInputSources/PixelSLinkDataInputSource.cc.d src/IORawData/SiPixelInputSources/src/PixelSLinkDataInputSource.cc -o tmp/el8_amd64_gcc12/src/IORawData/SiPixelInputSources/src/IORawDataSiPixelInputSources/PixelSLinkDataInputSource.cc.o
>> Compiling edm plugin src/IORawData/SiPixelInputSources/src/SealModule.cc
/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/gcc/12.3.1-40d504be6370b5a30e3947a6e575ca28/bin/c++ -c -DGNU_GCC -D_GNU_SOURCE -DTBB_USE_GLIBCXX_VERSION=120301 -DTBB_SUPPRESS_DEPRECATED_MESSAGES -DTBB_PREVIEW_RESUMABLE_TASKS=1 -DTBB_PREVIEW_TASK_GROUP_EXTENSIONS=1 -DBOOST_SPIRIT_THREADSAFE -DPHOENIX_THREADSAFE -DBOOST_MATH_DISABLE_STD_FPCLASSIFY -DBOOST_UUID_RANDOM_PROVIDER_FORCE_POSIX -DCMSSW_GIT_HASH='CMSSW_14_2_X_2024-10-04-1100' -DPROJECT_NAME='CMSSW' -DPROJECT_VERSION='CMSSW_14_2_X_2024-10-04-1100' -Isrc -Ipoison -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/cms/cmssw-patch/CMSSW_14_2_X_2024-10-04-1100/src -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/pcre/8.43-e34796d17981e9b6d174328c69446455/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/boost/1.80.0-477823d53efabc5118f199265eb7ab49/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/bz2lib/1.0.6-d065ccd79984efc6d4660f410e4c81de/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/libuuid/2.34-27ce4c3579b5b1de2808ea9c4cd8ed29/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/lcg/root/6.30.09-c59d6b036f1cbd6988c172ba319259f1/include -isystem/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/tbb/v2021.9.0-e785b749a0b6cb9c66dc1d78066210e0/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/xz/5.2.5-6f3f49b07db84e10c9be594a1176c114/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/zlib/1.2.11-1a082fc322b0051b504cc023f21df178/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/fmt/10.2.1-deba157dd76fed17ac8b42adc60ff53f/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/md5/1.0.0-5b594b264e04ae51e893b1d69a797ec6/include -I/cvmfs/cms-ib.cern.ch/sw/x86_64/nweek-02857/el8_amd64_gcc12/external/tinyxml2/6.2.0-68d511689c5dbd9a0d296bbb795fa052/include -O3 -pthread -pipe -Werror=main -Werror=pointer-arith -Werror=overlength-strings -Wno-vla -Werror=overflow -std=c++20 -ftree-vectorize -Werror=array-bounds -Werror=format-contains-nul -Werror=type-limits -fvisibility-inlines-hidden -fno-math-errno --param vect-max-version-for-alias-checks=50 -Xassembler --compress-debug-sections -Wno-error=array-bounds -Warray-bounds -fuse-ld=bfd -march=x86-64-v2 -felide-constructors -fmessage-length=0 -Wall -Wno-non-template-friend -Wno-long-long -Wreturn-type -Wextra -Wpessimizing-move -Wclass-memaccess -Wno-cast-function-type -Wno-unused-but-set-parameter -Wno-ignored-qualifiers -Wno-unused-parameter -Wunused -Wparentheses -Werror=return-type -Werror=missing-braces -Werror=unused-value -Werror=unused-label -Werror=address -Werror=format -Werror=sign-compare -Werror=write-strings -Werror=delete-non-virtual-dtor -Werror=strict-aliasing -Werror=narrowing -Werror=unused-but-set-variable -Werror=reorder -Werror=unused-variable -Werror=conversion-null -Werror=return-local-addr -Wnon-virtual-dtor -Werror=switch -fdiagnostics-show-option -Wno-unused-local-typedefs -Wno-attributes -Wno-psabi -Wno-error=unused-variable -DBOOST_DISABLE_ASSERTS -flto=auto -fipa-icf -flto-odr-type-merging -fno-fat-lto-objects -Wodr -fPIC -MMD -MF tmp/el8_amd64_gcc12/src/IORawData/SiPixelInputSources/src/IORawDataSiPixelInputSources/SealModule.cc.d src/IORawData/SiPixelInputSources/src/SealModule.cc -o tmp/el8_amd64_gcc12/src/IORawData/SiPixelInputSources/src/IORawDataSiPixelInputSources/SealModule.cc.o
src/IORawData/SiPixelInputSources/src/PixelSLinkDataInputSource.cc: In member function 'virtual bool PixelSLinkDataInputSource::setRunAndEventInfo(edm::EventID&, edm::TimeValue_t&, edm::EventAuxiliary::ExperimentType&)':
src/IORawData/SiPixelInputSources/src/PixelSLinkDataInputSource.cc:269:5: error: 'eventnumber' was not declared in this scope
  269 |     eventnumber = (m_data >> 32) & 0x00ffffff;
      |     ^~~~~~~~~~~
src/IORawData/SiPixelInputSources/src/PixelSLinkDataInputSource.cc:340:12: error: 'eventnumber' was not declared in this scope
  340 |   } while (eventnumber == m_currenteventnumber);
      |            ^~~~~~~~~~~
</pre>

## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Entering Package TopQuarkAnalysis/TopEventProducers
>> Entering Package TopQuarkAnalysis/TopHitFit
>> Entering Package TrackPropagation/RungeKutta
>> Entering Package TrackPropagation/SteppingHelixPropagator
>> Compile sequence completed for CMSSW CMSSW_14_2_X_2024-10-04-1100
gmake: *** [There are compilation/build errors. Please see the detail log above.] Error 1
+ eval scram build outputlog '&&' '(python3' /data/cmsbld/jenkins/workspace/ib-run-pr-tests/cms-bot/buildLogAnalyzer.py --logDir /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_14_2_X_2024-10-04-1100/tmp/el8_amd64_gcc12/cache/log/src '||' 'true)'
++ scram build outputlog
>> Entering Package CalibMuon/DTCalibration
Entering library rule at src/CalibMuon/DTCalibration/plugins
>> Compiling edm plugin src/CalibMuon/DTCalibration/plugins/DTCalibrationMap.cc
</pre>

--- a/IORawData/DTCommissioning/plugins/DTNewROS8FileReader.cc
+++ b/IORawData/DTCommissioning/plugins/DTNewROS8FileReader.cc
@@ -100,7 +100,7 @@ int DTNewROS8FileReader::fillRawData(Event& e,
     //2.- Get the HEADER  ============================================================================
     int datahead[numberEventHeadWords];
     for (int iih = 0; iih < numberEventHeadWords; iih++) {
-      nread = inputFile.read(dataPointer<int>(&datahead[iih]), ros8WordLenght);
+      inputFile.read(dataPointer<int>(&datahead[iih]), ros8WordLenght);
     }
 
     //3.- ROS DATA  &  4.- PU DATA (Trigger)   =======================================================

--- a/IORawData/SiPixelInputSources/src/PixelSLinkDataInputSource.cc
+++ b/IORawData/SiPixelInputSources/src/PixelSLinkDataInputSource.cc
@@ -259,8 +259,7 @@ bool PixelSLinkDataInputSource::setRunAndEventInfo(edm::EventID &id,
   // create product (raw data)
   buffers = std::make_unique<FEDRawDataCollection>();
 
-  //  uint32_t currenteventnumber = (m_data >> 32)&0x00ffffff;
-  uint32_t eventnumber = 0;
+  uint32_t eventnumber;
 
   do {
     std::vector<uint64_t> buffer;

--- a/IORawData/SiPixelInputSources/src/PixelSLinkDataInputSource.cc
+++ b/IORawData/SiPixelInputSources/src/PixelSLinkDataInputSource.cc
@@ -260,7 +260,7 @@ bool PixelSLinkDataInputSource::setRunAndEventInfo(edm::EventID &id,
   buffers = std::make_unique<FEDRawDataCollection>();
 
   //  uint32_t currenteventnumber = (m_data >> 32)&0x00ffffff;
-  uint32_t eventnumber = (m_data >> 32) & 0x00ffffff;
+  uint32_t eventnumber = 0;
 
   do {
     std::vector<uint64_t> buffer;

--- a/JetMETCorrections/MCJet/bin/ReadTree.cc
+++ b/JetMETCorrections/MCJet/bin/ReadTree.cc
@@ -119,7 +119,6 @@ int main(int argc, char **argv) {
     if (cut_ptmin && cut_dR && cut_njets) {
       ind_pt = getBin(ptGen, pt_vec);
       ind_eta = getBin(etaJet, eta_vec);
-      resp = 0.;
       if (UseRatioForResponse && ptGen > 0)
         resp = ptJet / ptGen;
       else

